Core/include/Acts/Propagator/detail/CovarianceEngine.hpp

```diff
@@ -24,6 +24,25 @@ namespace Acts {
 /// calculations are identical for @c StraightLineStepper and @c EigenStepper.
 /// As a consequence the methods can be located in a seperate file.
 namespace detail {
+using Covariance = std::variant<BoundSymMatrix, FreeSymMatrix>;
+using Jacobian =
+    std::variant<BoundMatrix, FreeToBoundMatrix, BoundToFreeMatrix, FreeMatrix>;
+
+/// @brief Constructs a jacobian to transform from (x,y,z,t,phi,theta,q/p) to
+/// (x,y,z,t,Tx,Ty,Tz,q/p)
+///
+/// @param [in] dir Direction vector
+///
+/// @return The jacobian
+ActsMatrixD<8, 7> jacobianDirectionsToAngles(const Vector3D& dir);
+
+/// @brief Constructs a jacobian to transform from (x,y,z,t,Tx,Ty,Tz,q/p) to
+/// (x,y,z,t,phi,theta,q/p)
+///
+/// @param [in] dir Direction vector
+///
+/// @return The jacobian
+ActsMatrixD<7, 8> jacobianAnglesToDirections(const Vector3D& dir);
```

Comment:
And the name of this function should be `jacobianDirectionsToAngles`?

Reply:
Fixed